src/Billable.php

```diff
@@ -40,26 +42,32 @@ public function charge($amount, array $options = [])
             $options['customer'] = $this->stripe_id;
         }
 
-        if (! array_key_exists('source', $options) && ! array_key_exists('customer', $options)) {
-            throw new InvalidArgumentException('No payment source provided.');
+        if (! array_key_exists('payment_method', $options) && ! array_key_exists('customer', $options)) {
+            throw new InvalidArgumentException('No payment method provided.');
         }
 
-        return StripeCharge::create($options, Cashier::stripeOptions());
+        $payment = new Payment(
+            StripePaymentIntent::create($options, Cashier::stripeOptions())
+        );
+
+        $payment->validate();
+
+        return $payment;
```

Comment:
Does this actually make the charge or does some other method need to be called on the returned object?

Reply:
This makes the actual charge because of the `'confirm' => true,` parameter.